app/src/pages/AbstractStreamEditPage/StreamModifier/index.jsx

```diff
@@ -0,0 +1,226 @@
+import React, { useMemo, useCallback, useEffect, useReducer, useRef } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useClient } from 'streamr-client-react'
+import cloneDeep from 'lodash/cloneDeep'
+import useModal from '$shared/hooks/useModal'
+import ValidationErrorProvider from '$shared/components/ValidationErrorProvider'
+import TransientStreamContext from '$shared/contexts/TransientStreamContext'
+import StreamModifierContext from '$shared/contexts/StreamModifierContext'
+import StreamModifierStatusContext from '$shared/contexts/StreamModifierStatusContext'
+import useStream from '$shared/hooks/useStream'
+import useInterrupt from '$shared/hooks/useInterrupt'
+import requirePositiveBalance from '$shared/utils/requirePositiveBalance'
+import getClientAddress from '$app/src/getters/getClientAddress'
+import InterruptionError from '$shared/errors/InterruptionError'
+import { networks } from '$shared/utils/constants'
+import DuplicateError from '$shared/errors/DuplicateError'
+import { useStreamSetter } from '$shared/contexts/StreamSetterContext'
+import useValidateNetwork from '$shared/hooks/useValidateNetwork'
+import routes from '$routes'
+import ConfirmExitModal from './ConfirmExitModal'
+import ChangeLossWatcher from './ChangeLossWatcher'
+import reducer, { initialState, Init, SetBusy, Modify } from './reducer'
+
+export default function StreamModifier({ children, onValidate }) {
+    const stream = useStream()
+
+    const [{ paramsModified, clean, busy }, dispatch] = useReducer(reducer, reducer(initialState, {
+        type: Init,
+        payload: stream,
+    }))
+
+    const firstRunRef = useRef(true)
+
+    useEffect(() => {
+        if (firstRunRef.current) {
+            firstRunRef.current = false
+            return
+        }
+
+        dispatch({
+            type: Init,
+            payload: stream,
+        })
+    }, [stream])
+
+    const paramsModifiedRef = useRef(paramsModified)
+
+    useEffect(() => {
+        paramsModifiedRef.current = paramsModified
+    }, [paramsModified])
+
+    const itp = useInterrupt()
```

Comment:
@tumppi, here's something interesting – `useInterrupt`. Remember `useIsMounted`? It's better! `useInterrupt.test` explains a lot. In short,

1. Let's say you have a `useCallback` with something asynchronous happening inside and it updates the UI on success.
2. You hit/call it 3 times and wait, you don't unmount the component.
3. Each resolved promise updates the UI (if we rely on `isMounted()`). 3 times in total, in order determined by the result of a race condition, which is something that you want very rarely.
4. `useInterrupt` along with a handler for `InterruptionError` can help you handle cancellation/skippage of the stale work.

Hope you can find a usecase for this in your work. Let's tweak it as we move forward.